scripts/ec2cleanup.py

```diff
@@ -218,8 +220,33 @@ def delete_queues(provider_mgmt, excluded_queues, output):
         logger.exception('Exception in %r', delete_queues.__name__)
 
 
+def delete_s3_bucket(provider_mgmt, bucket_name, output):
+    bucket_list = []
+    buckets = []
+    provider_name = provider_mgmt.kwargs['name']
+    try:
+        for bucket in provider_mgmt.list_s3_bucket():
+            if bucket_name in bucket:
```

Comment:
there is a lot of redundant code. please update it

Reply:
REQUIRED: this is still actual. 
example of optionization:
`
to_delete = [bucket for bucket in list_s3_bucket_names() if bucket_name in bucket]
deleted_buckets = provider_mgmt.delete_s3_buckets(bucket_names=to_delete)
`